### ISSUE TITLE
APIs from concerto-core are breaking for aliased models whose types are resolved.

### ISSUE THREAD

**Issue by sanketshevkar** [accordproject/concerto#972]
# Bug Report 🐛

The implementation for the resolution for the previous similar [issue](https://github.com/accordproject/concerto/issues/950) is leading to a few APIs in `concerto-core` and `concerto-cto` breakiing when they try to refer to the resolved type reference for the imported-aliased declarations.

Main issue is that when a model AST is resolved, we loose the reference to the aliased type names instead have the resolved type name, which was replaced.

## Expected Behavior

Resolved model should preserve both the resolved type names and the original type name as specified by the user.

## Current Behavior

Resolved model only replaces the type names used by the user with resolved type names (taking into account aliases).

## Steps to Reproduce

1. Model Decoration
If we try to decorate these models with given DCS, the process fails with error.
```
namespace accord.standard@1.0.0

concept BaseNDA {
    o Date executiondate
}

             +

namespace acme.legal.@1.0.0

import accord.standard@1.0.0.{BaseNDA as NewNDA}

concept BaseNDA extends NewNDA {
    o String person1
}

            +

{
    "commandSet": [
        {
            "type": "UPSERT",
            "target": {
              "namespace": "namespace acme.legal.@1.0.0",
              "declaration": "BaseNDA",
              "$class": "org.accordproject.decoratorcommands@0.3.0.CommandTarget"
            },
            "decorator": {
              "name": "TestMe",
              "arguments": [],
              "$class": "concerto.metamodel@1.0.0.Decorator"
            },
            "$class": "org.accordproject.decoratorcommands@0.3.0.Command"
          }
    ]
}

            =

    Error
```
2. Printing a resolved model from AST, where an declaration extending an aliased declaration would break
If we try to load these models, validate, resolve and print these models, the printing step would result in an error.
```
namespace accord.standard@1.0.0

concept BaseNDA {
    o Date executiondate
}

         +

namespace acme.legal.@1.0.0

import accord.standard@1.0.0.{BaseNDA as NewNDA}

concept BaseNDA extends NewNDA {
    o String person1
}

        =

Error
```
3. Print a resolved model from AST, where a property instantiated using an aliased declaration would change to the base declaration rather than the aliased declaration.
if we try to load these models, validate, resolve and print these models, the processes are successful but we loose reference to the aliased type which is imported in the property instantiation.
```
namespace accord.standard@1.0.0

concept BaseNDA {
    o Date executiondate
}

        +

namespace acme.legal.@1.0.0

import accord.standard@1.0.0.{BaseNDA as NewNDA}

concept NDA {
    o NewNDA newNDA
}

      =

import accord.standard@1.0.0.{BaseNDA as NewNDA}

concept NDA {
    o BaseNDA newNDA
}

```

### Desktop
 - OS: [e.g. macOS]
 - Browser: [e.g. Chrome, Safari]
 - Version: [e.g. 0.22.15]

## Detailed Description

The issue is that `resolveMetamodel` method (`called by ModelManager.toAst(true)`) was modified to replace aliased imports with the actual type names inside  TypeIdentifiers. This is useful for users processing the AST however it is lossy as it is difficult to recover the name as originally used by the user in the CTO file. In a subsequent call to `fromAst` the  resolved AST is used:
1. If the resolved type was not imported then a type not found error will occur
2. If the resolved imported type has the same name as a type  in the current namespace then a conflict will occur, leading to undefined behavior
3. If the resolved imported type is used a super class for a class with the same name, then a conflict will occur
4. Printer.toCTO has not been updated to recover the aliased name, so the roundtrip from [CTO with aliased import -> Resolved AST -> CTO] will be lossy, with aliased type names replaced with resolved type names

### ModelManager.getAst(true) should return: 
   1. A valid instance of the meta model 
   2. Namespaces should be added to TypeIdentifiers so that imports do not need to be inspected
   3. Set the `resolvedName` property to be the name of the type in its host namespace, irrespective of whether aliasing was used for the type or not.
   4. The `name` property should be the type name as specified in the CTO file

To get a valid fully-qualified name for a type identifier the caller should use `${namespace}.${resolvedName}` — this ensures that aliasing is taken into account.

### ModelManager.fromAst(ast) should return:

> No changes here. Undo the changes to fromAst.

### Printer.toCTO(ast) should return:

> No changes here. The name property should be used.

### Metamodel update

   1. Add an optional `resolvedName` to TypeIdentifier
   2. Republish concerto-metamodel
   3. Update models.accordproject.org

### Unit Tests

Alias an import and use wherever a type can be referenced:
- Super type
- Property
- Decorator
- Map key
- May value 

```
namespace child@1.0.0
concept Child {}
scalar MyString extends String
```

```
@test(Kid)
namespace parent@1.0.0

@test(Kid)
import child@1.0.0.{MyString as Str, Child as Kid}

@test(Kid)
concept Child extends Kid {
   @test(Kid)
   o Kid kid
   o Str str
}

@test(Kid)
map KidIndex {
   o Str
   @test(Kid)
   o Kid
}
```

Test that `parent@1.0.0` CTO file roundtrips through getAst(true) and fromAst(ast) and can be used with decorator command set.

**Comment by DS-AdamMilazzo** [comment]
I don't agree with the proposed implementation. In particular, I don't think the resolved type identifier's `name` should be the alias.

Previously, the rules were:
1. If there's a namespace, then namespace + name is the fully qualified name of the type.
2. Otherwise, the name must be resolved against the imports and declarations.

I think that should continue to be the case. Making namespace + name _not_ the fully qualified name of the type seems like a breaking change.

I don't see what the real problem is with the current format. Given that the imports still exist, the aliasing process can easily be reversed to recover the original names if desired. The only case when it wouldn't work is when a type is imported multiple times under different aliases - in that case, we could only pick one of them arbitrarily, but no matter which one we pick the result is semantically identical.
